docs/notebooks/model_config.pct.py

```diff
@@ -0,0 +1,209 @@
+
+# %%
+from copy import deepcopy
+
+from trieste.models.gpflux import DeepGaussianProcess, build_vanilla_deep_gp
+
+
+def build_gp_model(data):
+    variance = tf.math.reduce_variance(data.observations)
+    kernel = gpflow.kernels.Matern52(variance=variance, lengthscales=[0.2, 0.2])
+    model = GPR(data.astuple(), kernel, noise_variance=1e-5)
+    return model
+
+
+def build_dgp_model(data):
+    variance = tf.math.reduce_variance(data.observations)
+    dgp = build_vanilla_deep_gp(data.query_points, num_layers=2, num_inducing=100)
+    dgp.f_layers[-1].kernel.kernel.variance.assign(variance)
+    dgp.f_layers[-1].mean_function = gpflow.mean_functions.Constant()
+    dgp.likelihood_layer.likelihood.variance.assign(1e-5)
+    return dgp
+
+
+def run_experiment(model_config):
+    bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
+    result = bo.optimize(2, initial_data, model_config)
+    return result
+
+
+# configuration shared by all experiments, this is modified by each experiment condition
+basic_config = {
+    "model": gpflow_model,
+    "model_args": {
+        "num_kernel_samples": 100,
+    },
+    "optimizer": gpflow.optimizers.Scipy(),
+    "optimizer_args": {
+        "minimize_args": {"options": dict(maxiter=100)},
+    },
+}
+
+# here we specify our experiments
+experiment_conditions = [
+    {"model_args": {"num_kernel_samples": 500}},
+    {"model_args": {"num_kernel_samples": 1000}},
+    {"model": build_dgp_model(initial_data)},
+]
+
+results = []
+for exp in experiment_conditions:
+    model_config = deepcopy(basic_config)
+    for key in exp:
+        model_config[key] = exp[key]
+    results.append(run_experiment(model_config))
+
+
+# %% [markdown]
+# ## Registry of supported models
+#
+# Configuration dictionaries are made possible with the `ModelRegistry` that contains mapping between each model (e.g. GPflow or GPflux) and corresponding model wrapper and optimizer wrapper. All models that Trieste currently supports are registered there.
+#
+# You can add new models to the registry, in case you have custom models with which you wish to use the configuration dictionaries. Let see an example of this. We will register the `GPMC` model from GPflow that is currently not supported. You would likely need to create a new model wrapper and perhaps a new optimizer wrapper as well, but just for the sake of an example we will borrow here existing wrappers, `GaussianProcessRegression` and `Optimizer`.
+
+# %%
+from trieste.models import ModelRegistry
+
+# adding the GPMC model to the registry
+ModelRegistry.register_model(gpflow.models.GPMC, GaussianProcessRegression, Optimizer)
+
+# check if it has been registered
+print(gpflow.models.GPMC in ModelRegistry.get_registered_models())
```

Comment:
Is the print necessary?

```suggestion
gpflow.models.GPMC in ModelRegistry.get_registered_models()
```

Reply:
it is